lib/pbio/platform/ev3/platform.c

```diff
@@ -487,6 +488,12 @@ void ev3_panic_handler(int except_type, ev3_panic_ctx *except_data) {
     panic_putu32(except_data->spsr);
 
     panic_puts("\r\nSystem will now reboot...\r\n");
+
+    // Poke the watchdog timer with a bad value to immediately trigger it
+    // if it has already been configured. If it has *not* been configured,
+    // that means we are crashing in early boot, and we let the jump back
+    // to the reset vector take care of rebooting the system.
+    HWREG(SOC_TMR_1_REGS + TMR_WDTCR) = 0;
```

Comment:
Should we add a new `pbdrv_reset_reason_t` for panic?

Reply:
Do we need one? It currently reports as a watchdog reset

Reply:
I don't know if it is worth it or not. That is why I ask. :smile: 

There is a slight difference in that the true watchdog reset only happens if the event loops stops running while a panic is a different mode of failure. So could be potentially useful for getting info on the cause of a crash. But would anyone ever actually do that? Maybe not very likely.

Reply:
Doesn't seem too likely, since right now crash dumps aren't saved anywhere and you need to have the debug console connected to see them.

Perhaps we can change "watchdog" to a generic "fatal error"?

Reply:
OK, we'll leave it the way it is. I would prefer to stick with calling it watchdog and add a new reason if we need to differentiate.